src/components/[guild]/crm/CustomizeViewModal.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+  Box,
+  Checkbox,
+  HStack,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  useColorModeValue,
+} from "@chakra-ui/react"
+import { Column, Table } from "@tanstack/react-table"
+import { Modal } from "components/common/Modal"
+import { Reorder } from "framer-motion"
+import { DotsSixVertical } from "phosphor-react"
+import { useState } from "react"
+import { Member } from "./useMembers"
+
+type Props = {
+  isOpen: boolean
+  onClose: () => void
+  table: Table<Member>
+}
+
+const COLUMN_NAMES = {
+  identity: "Identity",
+  roles: "All roles",
+  hiddenRoles: "Hidden roles",
+  publicRoles: "Public roles",
+  joinedAt: "Joined at",
+}
+
+const CustomizeViewModal = ({ isOpen, onClose, table }: Props) => {
+  const columns = table.getAllColumns()
+  const [localOrder, setLocalOrder] = useState(() => transformToLocalOrder(table))
+
+  const setTableOrder = () => {
+    const newOrder = transformToTableOrder(localOrder)
+    // waiting for the animation to finish so the rerender doesn't cause a visible freeze
+    setTimeout(() => {
```

Comment:
I wonder if we could use `onAnimationComplete` instead of these timeouts 🤔

Reply:
Hmm I didn't know about that, I've tried it but it doesn't work on Reorder.Item unfortunately, so I've just extended the comment with info about it, ty!